include/drivers/dma.h

```diff
@@ -142,7 +147,11 @@ typedef void (*dma_callback_t)(const struct device *dev, void *user_data,
  *     linked_channel       [ 20 : 26 ] - after channel count exhaust will
  *                                        initiate a channel service request
  *                                        at this channel
- *     reserved             [ 27 : 31 ]
+ *     polling_mode         [ 27 ]      - whether dma uses async or polling mode
```

Comment:
Do we have some code or something here around `polling_mode`?

Reply:
it is in our local project currently, may get them upstreamed in future